graph/path/bellman_ford_moore.go

```diff
@@ -80,6 +80,77 @@ func BellmanFordFrom(u graph.Node, g graph.Graph) (path Shortest, ok bool) {
 	return path, true
 }
 
+// BellmanFordAllFrom returns a shortest-path tree for shortest paths from u to all nodes in
+// the graph g, or false indicating that a negative cycle exists in the graph. If the graph
+// does not implement Weighted, UniformCost is used.
+//
+// The time complexity of BellmanFordAllFrom is O(|V|.|E|).
+func BellmanFordAllFrom(u graph.Node, g graph.Graph) (path ShortestAlts, ok bool) {
+	if g.Node(u.ID()) == nil {
+		return ShortestAlts{from: u}, true
+	}
+	var weight Weighting
+	if wg, ok := g.(Weighted); ok {
+		weight = wg.Weight
+	} else {
+		weight = UniformCost(g)
+	}
+
+	nodes := graph.NodesOf(g.Nodes())
+
+	path = newShortestAltsFrom(u, nodes)
+	path.dist[path.indexOf[u.ID()]] = 0
+
+	// Queue to keep track which nodes need to be relaxed.
+	// Only nodes whose vertex distance changed in the previous iterations
+	// need to be relaxed again.
+	queue := newBellmanFordQueue(path.indexOf)
+	queue.enqueue(u)
+
+	// The maximum number of edges in a graph is |V| * (|V|-1) which is also
+	// the worst case complexity.
+	// If the queue-loop has more iterations than the amount of maximum edges
+	// it indicates that we have a negative cycle.
+	maxEdges := len(nodes) * (len(nodes) - 1)
+	var loops int
```

Comment:
nit: I would move this just before the for loop.

Reply:
OK. The rationale for it being where it is is to explain the reason for `loops`' existence. I'll send another PR after this one changing it here and in the original code after we have merged this.

Reply:
If you made it on purpose, then it's fine to leave as is, of course.